third-culture-software/bhima
# Report for show needed inventory scans for a required inventory scan

Added reports folder the under Assets Management menu item.
Added "Needed Inventory Scans" page in the new folder.  This allows you to select one of the "Required Inventory Scans" and see the all assets that are covered by the scan, as well as scanned and unscanned items.

Note that Required Inventory Scans require a start date and end date and "is_asset".  These are used directly from the chosen

Closes https://github.com/IMA-WorldHealth/bhima/issues/6468
Closes https://github.com/IMA-WorldHealth/bhima/issues/6481
(I need to make sure the "Additions Needed" are done or covered in a new issue).

**TESTING**
- Use bhima_test
- Go to Assets Management > Asset Scans  Registry
  - Note that the "created_date" on one of the asset scans has been pushed back in time.  You can use this to test whether the required start and end dates are observed in the report.
  - Edit an asset to add a reference_number
  - Try setting a Search filter for this reference number
- Go to Assets Management > Required Inventory Scans
  - Create new scan(s).  Either create several scans to check the filtering behavior or go back and edit the Required Inventory Scan item to change the necessary settings
- Go to Assets Management > Reports > Needed Inventory Scans
  - Select the Required Inventory Scan and scan status and try the report
  - Please try a variety of options to verify correct filtering behavior of 
    - From Required Inventory Scan:
        - start date
        - end date
        - is_asset
        - reference_number
    - Scan type:  all, scanned, unscanned

- Please double-check with a production data set to make sure that the migration file is correct.
